### PR TITLE
Asset Server caching

### PR DESCRIPTION
**Pull request by Atlante45**
Enable local caching on the client side for resources from the asset server.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2365fb511ecd715b29260880c487b7af515f4c7f available at: https://gist.github.com/9662d212050501d4cd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5806.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5806.dmg

File: libraries/networking/src/AssetClient.cpp [hunk 1]
```diff
@@ -40,9 +45,26 @@ AssetClient::AssetClient() {
...
+    
+    // Setup disk cache if not already
+    QNetworkAccessManager& networkAccessManager = NetworkAccessManager::getInstance();
+    if (!networkAccessManager.cache()) {
+        QString cachePath = QStandardPaths::writableLocation(QStandardPaths::DataLocation);
+        QNetworkDiskCache* cache = new QNetworkDiskCache();
+        cache->setMaximumCacheSize(MAXIMUM_CACHE_SIZE);
+        cache->setCacheDirectory(!cachePath.isEmpty() ? cachePath : "interfaceCache");
+        networkAccessManager.setCache(cache);
+        qCDebug(asset_client) << "AssetClient disk cache setup.";
```
Comment (birarda) [inline libraries/networking/src/AssetClient.cpp]: still want this in here?

Comment (birarda) [inline libraries/networking/src/AssetClient.cpp]: If you do I would add some more information (MAXIMUM_CACHE_SIZE + directory)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88d073b4ac1e4a44ea7b4b71297145335e5af751 available at: https://gist.github.com/9662d212050501d4cd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5806.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5806.dmg

**Pull request merged**